app/src/main/java/com/asfoundation/wallet/ui/backup/entry/BackupWalletPresenter.kt

```diff
@@ -0,0 +1,66 @@
+package com.asfoundation.wallet.ui.backup.entry
+
+import com.asfoundation.wallet.ui.balance.BalanceInteractor
+import com.asfoundation.wallet.util.CurrencyFormatUtils
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+import java.util.concurrent.TimeUnit
+
+class BackupWalletPresenter(private val balanceInteractor: BalanceInteractor,
+                            private val view: BackupWalletFragmentView,
+                            private val data: BackupWalletData,
+                            private val navigator: BackupWalletNavigator,
+                            private val currencyFormatUtils: CurrencyFormatUtils,
+                            private val disposables: CompositeDisposable,
+                            private val dbScheduler: Scheduler,
+                            private val viewScheduler: Scheduler) {
+
+  fun present() {
+    handleBackupClick()
+    handlePasswordTextChange()
+    retrieveStoredBalance()
+  }
+
+  private fun handlePasswordTextChange() {
+    disposables.add(view.onPasswordTextChanged()
+        .observeOn(viewScheduler)
+        .doOnNext {
+          if (it.empty) view.clearErrors()
+          if (!it.match) view.disableButton()
+          if (it.match && !it.empty) {
+            view.clearErrors()
+            view.enableButton()
+          }
+        }
+        .debounce(700, TimeUnit.MILLISECONDS, viewScheduler)
+        .doOnNext { if (!it.match && !it.empty) view.showPasswordError() }
+        .subscribe({}, { it.printStackTrace() }))
+  }
+
+  private fun handleBackupClick() {
+    disposables.add(view.getBackupClick()
+        .doOnNext {
+          view.hideKeyboard()
+          val password = if (it.wantsPassword) it.password else ""
+          navigator.showBackupCreationScreen(data.walletAddress, password)
+        }
+        .subscribe({}, { it.printStackTrace() }))
+  }
+
+  private fun retrieveStoredBalance() {
```

Comment:
I would maybe rename this method to initializeView or similar, because it is the effect on the view that matters, I think. "retrieveStoredBalance" almost sounds like a getter.